packages/mui-varnish/src/CopyToClipboardButton.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { useState } from 'react';
+import Snackbar from '@mui/material/Snackbar';
+import IconButton from '@mui/material/IconButton';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+
+interface Props {
+  text?: string;
+  children:
+    | string
+    | number
+    | boolean
+    | React.ReactElement<any, string | React.JSXElementConstructor<any>>
+    | React.ReactFragment
+    | React.ReactPortal;
+}
+function CopyToClipboardButton({ text, children }: Props) {
+  const [open, setOpen] = useState(false);
+  const handleClick = () => {
+    setOpen(true);
+    navigator.clipboard.writeText(text ? (text as string) : children.toString());
+  };
+
+  return (
+    <React.Fragment>
+      {children}
+      <IconButton color="primary" size="small" onClick={handleClick}>
+        <ContentCopyIcon fontSize="inherit" />
+      </IconButton>
+      <Snackbar
+        open={open}
+        onClose={() => setOpen(false)}
+        autoHideDuration={2000}
```

Comment:
This seems like something a user might want to toggle, maybe it should be a prop with a default?